### DISCUSSION TITLE
Godot's code editor to support any language's features

### DISCUSSION THREAD

**Discussion by dmitryuck** [godotengine/godot-proposals#7893] [Editor]
### Describe the project you are working on

Godot Editor

### Describe the problem or limitation you are having in your project

Developers using GDExtension to integrate any language support for scripting in Godot
But those languages are not supported by Godot's code Editor on a level as GDscript

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

The idea is to give developers an opportunity to support any language features like highlighting, syntax, auto-completing, and formatting in Godot's code editor, so users will not need to use an external code editor for their languages.
For this I believe should be implemented an API like Language Server Extension in vscode, Through this API they can connect a compiler of a particular language and communicate with Godot's code Editor.
Correct me if  I'm wrong, but I believe you got the idea)

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

The Godot's code editor will identify the language by script file extension and then communicate with the compiler through API and we have a full code support for any language

### If this enhancement will not be used often, can it be worked around with a few lines of script?

It's difficult task

### Is there a reason why this should be core and not an add-on in the asset library?

This is support from Godot's Editor

**Comment by MeisterDerMagie** [comment]
This would result in a code editor that has only a fraction of the features of established IDEs. Most developers would continue to use external editors. So this would lead to a feature that needs to be maintained, but has little application.
